lambda-buffers-proto/compiler.proto

```diff
@@ -636,45 +636,56 @@ message ProtoParseError {
 // Kind checking errors.
 message KindCheckError {
 
-  // Error referring to an unbound term. This usually means that the term was
-  // not defined.
-  message UnboundTermError {
-    TyName ty_name = 1;
-    VarName var_name = 2;
+  // Unbound variable ty_var detected in term ty_def.
+  message UnboundTyVarError {
+    TyDef ty_def = 1;
+    TyVar ty_var = 2;
+    ModuleName module_name = 3;
+  }
+
+  // Unbound type reference ty_ref detected in term ty_def.
+  message UnboundTyRefError {
+    TyDef ty_def = 1;
+    TyRef ty_ref = 2;
+    ModuleName module_name = 3;
   }
 
-  // Unification has failed - type is incorrectly defined. Error reads as
-  // follows:
-  // > In ty_name definition an error has occurred when trying to unify kind
-  // > ty_kind_1 with ty_kind_2.
+  // In ty_def an error has occurred when trying to unify kind ty_kind_lhs 
+  // with ty_kind_rhs.
   //
-  // FIXME(cstml): add source of constraint to the error such that user can see
+  // FIXME(cstml): Add source of constraint to the error such that user can see
   // where the constraint was generated - therefore where the error precisely
   // is.
   message ImpossibleUnificationError {
-    TyName ty_name = 1;
-    Kind ty_kind_1 = 2;
-    Kind ty_kind_2 = 3;
+    TyDef ty_def = 1;
+    Kind ty_kind_lhs = 2;
+    Kind ty_kind_rhs = 3;
+    ModuleName module_name = 4;
   }
 
-  // Inifinitely recursive term detected in definition ty_name.
+  // A cyclic term was encountered. Infinite terms like this are not acceptable,
+  // and we do not support them. We could not construct infinite "a = ty_name a"
+  // error.
   message RecursiveKindError {
-    TyName ty_name = 1;
+    TyDef ty_def = 1;
+    ModuleName module_name = 2;
   }
```

Comment:
If you could just elaborate on how this can be triggered by the API users that would be great. It says RecursiveKind but in the doc it says `cyclic term` which would mean `cyclic type` right?

Reply:
It means the same thing in this context.